Firestore/core/src/firebase/firestore/model/maybe_document.h

```diff
@@ -41,6 +41,7 @@ class MaybeDocument {
     Unknown,
     Document,
     NoDocument,
+    UnknownDocument,
```

Comment:
Since you are adding type here, could you add document for each of the type? In particular, it helps reader differing `Unknown` and `UnknownDocument`.

Reply:
I'd prefer not to add documentation for each type, since that can already be found in the subclasses themselves. So instead, I've added "See the docstrings of the subclasses for details." to the docstring of the enum itself.

However, "Unknown" isn't a subclass. It used to be used by MaybeDocument in situations where a MaybeDocument was created directly. That's no longer possible since MaybeDocument is now abstract. It's also used as a placeholder for an invalid value that indicates a pretty serious problem with serialization. I've added a docstring for this value.

I've also added a TODO to investigate removing it entirely; I suspect the serializers can do without it.